lib/prediction/model.js

```diff
@@ -0,0 +1,370 @@
+      return;
+    }
+
+    var results = {
+      data: resp.dataDescription || {},
+      model: resp.modelDescription || {}
+    };
+
+    callback(null, results, resp);
+  });
+};
+
+/**
+ * Create a writable stream to train the model with new data.
+ *
+ * This is a wrapper around {module:prediction/model#train}.
+ *
+ * @param {string|number} label - The output value, either a regression or class
+ *     label.
+ * @return {WritableStream}
+ *
+ * @example
+ * var stream = model.createWriteStream('english');
+ *
+ * stream
+ *   .on('error', function(err) {
+ *     // Uh oh, an error occurred!
+ *   })
+ *   .on('finish', function() {
+ *     // The model will now be processing the new data.
+ *   });
+ *
+ * stream.write('Hello, this is an English sentence!');
+ * stream.write('...yeah, I assumed that when I was able to read it.');
+ * stream.end();
+ */
+Model.prototype.createWriteStream = function(label) {
+  var self = this;
+  var writeStream = streamEvents(pumpify());
+
+  writeStream.once('writing', function() {
+    var template = JSON.stringify({
+      output: label,
+      csvInstance: [[]]
+    }).split('[]');
+
+    var requestStream = self.request({
+      method: 'PUT',
+      uri: '',
+      headers: {
+        accept: 'application/json',
+        'content-type': 'application/json'
+      }
+    });
+
+    requestStream.on('response', function(resp) {
+      writeStream.emit('response', resp);
+    });
+
+    writeStream.setPipeline([
+      through({ encoding: 'utf-8' }),
+      JSONStream.stringify(template[0], ',', template[1]),
+      requestStream
+    ]);
+  });
+
+  return writeStream;
+};
+
+/**
+ * Query the stored data in your model.
+ *
+ * @resource [Trainedmodels:predict API Documentation]{@link https://cloud.google.com/prediction/docs/reference/v1.6/trainedmodels/predict}
+ *
+ * @param {string|string[]} input - Input feature(s) to query on the model.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An API error.
+ * @param {?object} callback.results - An organized set of results from the
+ *     prediction.
+ * @param {string} callback.results.winner - The matching result's label.
+ * @param {object[]} callback.results.scores - Sorted from highest-to-lowest,
+ *     all of the results that matched the query.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * //-
+ * // Assuming this is a model used to return a language name that matches any
+ * // given input, this will ask for the prediction results of an english
+ * // sentence.
+ * //-
+ * model.query('Hello', function(err, results) {
+ *   if (!err) {
+ *     // `results.winner` == 'english'
+ *     // `results.scores` == [
+ *     //   {
+ *     //     label: 'english',
+ *     //     score: 1
+ *     //   },
+ *     //   ...other results and their scores are listed here.
+ *     // ]
+ *   }
+ * });
+ */
+Model.prototype.query = function(input, callback) {
```

Comment:
Just a case where I went with the logical action name. It felt strange using the action "predict", since the user isn't predicting anything, they're asking for a prediction. 

Throughout the docs, they refer to it as "querying": https://cloud.google.com/prediction/docs/hello_world